Add Metafield Resource

aryy opened the pull request:
* Adds the general Metafield resource (e.g. `client.Metafield.List(...)`)
* Adds metafields to Product resource (e.g. `client.Product.ListMetafields(...)`)

davidtonghelix committed:
Added Customer.Create() and unit test

davidtonghelix committed:
Revert oauth.go

davidtonghelix committed:
Merge pull request #1 from myhelix/dt-customer-create

Adding Customer.Create

Commit:
Add support for Metafield API

aryy committed:
Merge pull request #2 from myhelix/con-829-aa-add-metafield

Add support for Metafield Resource

Commit:
Add metafields to product resource

[CON-829]

Commit:
Change structure for accessing resource metafields

[CON-829]

aryy committed:
Merge pull request #3 from myhelix/con-829-add-metafields-to-product-resource

Add Metafields to Product resource

aryy committed:
Merge branch 'master' into master

Commit:
Fix formatting

Commit:
Add Create to OrderService.

* omitempty all the things! Order can accept pretty minimal information.

dlebech commented on goshopify.go:
```diff
@@ -201,11 +203,19 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
```
Why did you make changes to this file? This seems to be unrelated to the main PR, so could you revert it for now?

dlebech reviewed (changes requested):
Thanks for the PR @aryy ! Looks very useful. I only have one comment on a piece of code that probably shouldn't be changed, and if you could fix the Travis error, that would be great. It's a small format problem in util_test and only seems to be a problem on Golang 1.10. The [error from Travis](https://travis-ci.org/getconversio/go-shopify/jobs/385965742):
```
./util_test.go:79: Errorf format %s has arg c.resourceID of wrong type int
```

Commit:
Address PR feedback

- Update format type from string to int
- Revert Unmarshal change

gir committed:
Merge pull request #4 from myhelix/ky-create-order

Add Create to OrderService.

Commit:
Update NoteAttribute to Marshal to lower case

Shopify does not expect/accept line_items properties (NoteAttributes) beginning with upper case.

gir committed:
Merge pull request #6 from myhelix/ky-fix-property-json

Update NoteAttribute to Marshal to lower case

gir commented:
Sorry, we'll point the PR at another branch and open another PR for the order-related changes as well.

Commit:
Remove files

Commit:
Merge branch 'master' of https://github.com/myhelix/go-shopify into aa-minor-changes

aryy committed:
Merge pull request #5 from myhelix/aa-minor-changes

Address PR feedback

aryy commented:
Closing in favor of https://github.com/getconversio/go-shopify/pull/90. This PR inadvertently pointed to our master branch.
